conda-incubator/conda-store
# restart workers after executing 10 tasks to mitigate memory leaks

Fixes #https://github.com/nebari-dev/nebari/issues/2418




## Description



This pull request:

- restarts celery workers after they execute 10 tasks (docs [here](https://docs.celeryq.dev/en/stable/userguide/workers.html#max-tasks-per-child-setting)).  This mitigates memory leaks which are occurring.

## Pull request checklist




## Additional information



## How to test



@trallard  edited from: https://github.com/conda-incubator/conda-store/pull/840#issuecomment-2207828243

To test, you could check out this branch, then run conda standalone [conda.store/conda-store/how-tos/install-standalone](https://conda.store/conda-store/how-tos/install-standalone) or you can run it in [docker containers](docker-compose up --build -d as explained [here](https://conda.store/community/contribute/contribute-code)). After you startup, you can visit localhost:8080/conda-store/admin and rebuild the environment it comes with a few times. If you watch the docker memory usage (docker stats) you'll see that the memory usage does not continue to climb with each new build as it does without this option set.
